miden-tx/src/host/mod.rs

```diff
@@ -163,6 +164,25 @@ impl<A: AdviceProvider, T: TransactionAuthenticator> TransactionHost<A, T> {
         Ok(())
     }
 
+    fn on_note_add_asset<S: ProcessState>(
+        &mut self,
+        process: &S,
+    ) -> Result<(), TransactionKernelError> {
+        //# => [ASSET, note_ptr]
+        let note_ptr: MemoryAddress = process
+            .get_stack_item(0)
+            .try_into()
+            .map_err(TransactionKernelError::MalformedNotePointer)?;
+        let asset = Asset::try_from(process.get_stack_word(0))
+            .map_err(TransactionKernelError::MalformedAsset)?;
+
+        let note_builder = self.output_notes.get_mut(&note_ptr).expect("note not found");
+
+        note_builder.add_asset(asset).expect("failed to add asset to note");
+
```

Comment:
I think these `.expect()` should be errors too.

Reply:
should that suffice? The error should be propagated

```
note_builder.add_asset(asset)?;
```

we have the same in `on_creata_note`

Reply:
Yep - I think this works. I would still convert `expect("note not found") into an error though.